File: models/Utilisateur.js
const mongoose = require('mongoose');
const bcrypt   = require('bcryptjs');

const utilisateurSchema = new mongoose.Schema({
  nom:                 { type: String, required: true, maxlength: 100 },
  prenom:              { type: String, maxlength: 100, default: '' },
  email:               { type: String, required: true, unique: true, lowercase: true },
  motDePasse:          { type: String, required: true },
  role:                { type: String, enum: ['AGENT', 'ADMIN'], default: 'AGENT' },
  isActif:             { type: Boolean, default: true },
  telephone:           { type: String, maxlength: 30, default: '' },
  departement:         { type: String, maxlength: 100, default: '' },
  poste:               { type: String, maxlength: 100, default: '' },
  niveauAccreditation: { type: String, maxlength: 50, default: '' },
  dateArrivee:         { type: Date, default: null },
}, { timestamps: true });

utilisateurSchema.pre('save', async function (next) {
  if (!this.isModified('motDePasse')) return next();
  this.motDePasse = await bcrypt.hash(this.motDePasse, 10);
  next();
});

utilisateurSchema.methods.verifierMotDePasse = function (motDePasse) {
  return bcrypt.compare(motDePasse, this.motDePasse);
};

module.exports = mongoose.model('Utilisateur', utilisateurSchema);

File: models/utilisateur.py
from typing import Literal
from datetime import datetime
from beanie import Document
from passlib.context import CryptContext

pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")


class Utilisateur(Document):
    nom: str
    email: str
    mot_de_passe: str
    role: Literal["AGENT", "ADMIN"] = "AGENT"
    is_actif: bool = True
    created_at: datetime = datetime.utcnow
    updated_at: datetime = datetime.utcnow

    class Settings:
        name = "utilisateurs"

    # ───────── HASH PASSWORD ─────────
    def set_password(self, password: str):
        self.mot_de_passe = pwd_context.hash(password)

    # ───────── VERIFY PASSWORD (OFFICIEL) ─────────
    def check_password(self, password: str) -> bool:
        return pwd_context.verify(password, self.mot_de_passe)

    # ───────── ALIAS COMPAT BACKEND (IMPORTANT) ─────────
    def verifier_mot_de_passe(self, password: str) -> bool:
        return self.check_password(password)